ocaml/xapi/xapi_sr.ml

```diff
@@ -424,22 +424,47 @@ let maybe_copy_sr_rrds ~__context ~sr =
     with Rrd_interface.Rrdd_error(Archive_failed(msg)) ->
       warn "Archiving of SR RRDs to stats VDI failed: %s" msg
 
+(* Removes the SR and dependent objects from the database (VDIs and PBDs).
+   This function is meant to be used only inside this module.
+   Also removes the associated metrics from xcp-rrdd until SR metrics are
+   correctly reported as SR ones (CA-325582)
+ *)
+let _destroy ~__context ~sr =
+  (* CA-325582: Since the pbds are not mounted anymore there are no more
+     metrics being reported, the current ones can be deleted. *)
+  let short_uuid = String.sub (Db.SR.get_uuid ~__context ~self:sr) 0 8 in
+  let is_sr_metric = Astring.String.is_suffix ~affix:short_uuid in
+
+  (* SR data sources are currently stored in memory as host ones.
+     Pick the ones that match the short uuid and remove them,
+     this prevents these metrics from being archived *)
+  Rrdd.query_possible_host_dss ()
```

Comment:
These are local calls to xcp-rrdd? Then I assume that they must be run on the "SR master". Keep in mind that `SR.forget` is not forwarded and always runs on the pool master. If the SR master is not the pool master, it will not work.

Reply:
So where exactly are these metrics? On every host that has a PBD for the SR? Or just on the SR master?

Reply:
The metrics are stored in every host that has a PBD for the SR